### DISCUSSION TITLE
Command-line interfaces

### DISCUSSION THREAD

**Discussion by joaomcteixeira** [THGLab/MCSCE#3] [Implementations]
To our current understanding of the project, which and how many command-line interfaces will be required? @JerryJohnsonLee , you have been using the original code quite intensively, do you foresee the need for several `CLIs` that would grant fine control to the user on the different tasks that the program will perform? Or is one general `CLI` enough?

**Comment by JerryJohnsonLee** [comment]
Currently we have a single CLI interface (the entrance python script) that sets up the calculations and generates structure ensemble. From a design perspective I would say maybe energy calculations and entropy calculations can be two different interfaces that is useful.

**Reply by joaomcteixeira** [reply]
Do you mean two `CLI` that would calculate energy and entropy on a battery of PDB files saved in the hardisk? It sounds good to me.

I can prepare the template code to accommodate different `CLIs` and then you could populate those `CLIs` with the pertinent options. Btw, which options would you envisage the `CLIs` having? In general terms, does not need to be a detailed list.

**Reply by JerryJohnsonLee** [reply]
Yes, or maybe just preparing python functions for these calculations is also enough? So we will create a "protein" class that stores the coordinates of a parsed structure from pdb/mmcif right? We should make the functions support either the string of the source structure file or that "protein" object

**Reply by joaomcteixeira** [reply]
Yes, that is right. So those are two different things, though. One thing is the functions that we will use internally to build the workflow and another thing is the `CLI` that is that the user would need to run a battery of energy calculations on a set of PDBs that are stored in a folder. Do you think the latter will be necessary?

**Reply by JerryJohnsonLee** [reply]
Probably not at highest priority right now
